apps/backend/test/mocks/classes/transaction_mock.ts

```diff
@@ -1,43 +1,33 @@
 import { faker } from '@faker-js/faker';
 import { format } from 'date-fns';
-import { Payment } from './payment_mock';
-import { BaseData } from '../types/interface';
+import { PaymentMock } from './payment_mock';
+import { PaymentEntity } from './../../../src/transaction/entities/payment.entity';
 import { Ministries } from '../../../src/constants';
-import { LocationEntity } from '../../../src/location/entities/master-location-data.entity';
+import { DateRange } from '../../../src/constants';
+import { LocationEntity } from '../../../src/location/entities';
 import { TransactionEntity } from '../../../src/transaction/entities';
 
 /*eslint-disable */
-export class Transaction extends TransactionEntity {
-  transaction_id: string;
-  program: Ministries;
-  location: LocationEntity;
-  transaction_date: string;
-  total_payment_amount: number;
-  fiscal_close_date: string;
-  payments: Payment[];
-
-  constructor(data: BaseData, payments: Payment[]) {
+export class TransactionMock extends TransactionEntity {
+  constructor(
+    dateRange: DateRange,
+    program: Ministries,
+    location: LocationEntity,
+    payments: PaymentMock[]
+  ) {
     super();
+
     this.transaction_id = `${faker.datatype.uuid()}`;
-    this.program = data.program;
-    this.location = data.location;
     this.fiscal_close_date = format(
-      faker.date.between(
-        `${data.dateRange.from_date}`,
-        `${data.dateRange.to_date}`
-      ),
-      'yyyy-MM-dd'
-    );
-
-    this.transaction_date = format(
-      faker.date.between(
-        `${data.dateRange.from_date}`,
-        `${data.dateRange.to_date}`
-      ),
+      faker.date.between(`${dateRange.from_date}`, `${dateRange.to_date}`),
       'yyyy-MM-dd'
     );
-    this.total_payment_amount = payments.reduce(
-      (acc: any, payment: Payment) => (acc += payment.amount),
+    this.source_id = program;
+    this.location_id = location.location_id;
+    this.transaction_date = dateRange.to_date;
+    this.transaction_time = '00:00:00';
```

Comment:
Any value in randomizing / fakering this?

Reply:
They're randomized and using faker, but have a specific set of rules in order for them to be 'reconcile-able'.

ie: if generating random cash deposits and cash transactions/payments to test (happy path) reconciliation then we need to make sure the sum of payments per fiscal_close_date === the deposit_amt_cdn for a deposit date, and also that the payment methods and locations match. 

The purpose of the mocks was to avoid writing it out for each unit test, but I am 100% open to suggestions for a better approach :)